src/frontend/src/binder/relation/table_or_source.rs

```diff
@@ -137,7 +155,8 @@ impl Binder {
                         self.resolve_source_relation(&source_catalog.clone(), as_of, true)?
                     } else if let Ok((table_catalog, schema_name)) = self
                         .catalog
-                        .get_created_table_by_name(db_name, schema_path, table_name)
+                        .get_any_table_by_name(db_name, schema_path, table_name)
+                        && (bind_creating_relations || table_catalog.is_created())
```

Comment:
Preserves the original semantics + accepts any table if `bind_creating_relations` is set.

Reply:
nit: `if bind_creating_relations { get_any_table_by_name } else { get_created_table_by_name }` might be slightly better. Or just add `bind_creating_relations` param to `get_table_by_name`

Reply:
> if bind_creating_relations { get_any_table_by_name } else { get_created_table_by_name }

This makes the code rather unreadable. We end up with something like:
```
if let Ok(catalog) = { if bind_creating_relations { get_any_table_by_name } else { get_created_table_by_name } } {
 // body of branch
}
```

> Or just add bind_creating_relations param to get_table_by_name

Took a look at the implementation. If we add this to `get_any_table_by_name` as a parameter, it breaks the semantics of the "any" part. We will have to introduce a new function instead. `fn get_table_by_name(..., bind_creating_relations: bool)`. I will add it.

Reply:
Fixed https://github.com/risingwavelabs/risingwave/pull/22205/commits/3881c3f09ebd7c617f35e6081a2aec1e859b98cf